addons/sale_management/models/sale_order.py

```diff
@@ -208,6 +207,24 @@ class SaleOrderOption(models.Model):
     quantity = fields.Float('Quantity', required=True, digits='Product Unit of Measure', default=1)
     sequence = fields.Integer('Sequence', help="Gives the sequence order when displaying a list of optional products.")
 
+    def _update_price_discount(self):
+        for option in self:
+            if not option.product_id:
+                continue
+            price = option.product_id.lst_price
+            discount = 0
+            pricelist = option.order_id and option.order_id.pricelist_id
+            if pricelist:
+                pricelist_price = pricelist.with_context(uom=option.uom_id.id).get_product_price(
+                    option.product_id, 1, False)
+
+                if pricelist.discount_policy == 'without_discount' and price:
+                    discount = max(0, (price - pricelist_price) * 100 / price)
+                else:
+                    price = pricelist_price
+
+            option.write({'price_unit': price, 'discount': discount})
```

Comment:
```suggestion
        values = option._get_values_to_add_to_order()
        new_sol = option.env['sale.order.line'].new(values)
        new_sol._onchange_discount()
        option.discount = new_sol.discount
        if option.order_id.pricelist_id and option.order_id.partner_id:
            option.price_unit = new_sol._get_display_price(product)
```

Why didn't you copy the logic below ?

Reply:
I looked to see if there was other place which my new function could replace but glossed over this as there was not the `without_discount` line i searched for :facepalm: 
Should be better now :D